src/control_interface/src/command_register.rs

```diff
@@ -0,0 +1,43 @@
+use core::ffi::c_void;
+use std::{collections::HashMap, ffi::CString};
+static NUM_COMMANDS: usize = 32;
+
+/// <div rustbindgen nocopy></div>
+/// <div rustbindgen opaque></div>
+#[repr(C)]
+pub struct RegisteredCommand {
+    object: *mut c_void,
+    action: *mut c_void,
+}
```

Comment:
Registered command may not need an 'object' but we might want to have a 'closure' object that contains the populated struct of the command payload as well as the 'action', i.e. function pointer.